### ISSUE TITLE
Browser Crash bug with big tags (ajax endpoint)

### ISSUE THREAD

**Issue by atakanuludag** [woothemes/woocommerce#31851]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Hi,
I am using the latest wordpress and woocommerce versions.

- Woocommerce version: 6.1.1
- Wp version: 5.9

Previous large tags crash the browser when adding products.

**Example api;**
`/wp-admin/admin-ajax.php?action=ajax-tag-search&tax=product_tag&q=lg`

![image](https://user-images.githubusercontent.com/6053742/152769727-b237ed6f-0911-476c-acfc-c2416e1d52b4.png)

![image](https://user-images.githubusercontent.com/6053742/152770559-b0a2debb-1186-4806-acd2-56a44e4c59d9.png)

### Expected behavior

Browser not crashing and ajax response limit

### Actual behavior

Browser crash and endpoint no limit

### Steps to reproduce

1. Admin panel
2. Products
3. New Products
4. Tags

### WordPress Environment

![image](https://user-images.githubusercontent.com/6053742/152769716-99cd62f7-4709-4898-bbc4-4cd4fb8c7c43.png)

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by woocommercebot** [comment]
Hi @atakanuludag,

  Thank you for opening the issue! It requires further feedback from the WooCommerce Core team.

  We are adding the `needs developer feedback` label to this issue so that the Core team could take a look.

  Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by atakanuludag** [comment]
@woocommercebot when will it be resolved?

**Comment by barryhughes** [comment]
Hi @atakanuludag,

I'll note first of all that this functionality is inherited from WordPress itself, so it may be most appropriate to deal with it at that level.

> Steps to reproduce
>
> 1. Admin panel
> 2. Products
> 3. New Products
> 4. Tags

So I think you are saying we should go to the Product Tags meta box in the product editor to see this:

<img width="296" alt="product-editor-tags" src="https://user-images.githubusercontent.com/3594411/157562635-5b81c985-e2e4-4520-9ef9-5ba3b2d0fc11.png">

What I'm not clear on, is how much data is being returned (it's not super clear from the screenshot you shared where one tag starts and another ends). Is it the case that you have an extremely large number of tags beginning with `LG`? How many, and what is the average length?

Thanks!

**Comment by atakanuludag** [comment]
Hi,
Yes true.
But the problem seems to be solved. I think it's fixed with the new update :)
Number of tags named **LG**:  **3,505**

But I still think there should be a length limit. Because it shows all matching tags as autocomplete item;
![image](https://user-images.githubusercontent.com/6053742/158155903-d3d002d9-2f2f-4b01-876a-3942c7bbe028.png)

**Comment by barryhughes** [comment]
> But the problem seems to be solved. I think it's fixed with the new update :)

Great! 🎉 

> But I still think there should be a length limit. Because it shows all matching tags as autocomplete item

Yep, that's a good call.

This is essentially an 'upstream' problem (we are simply using WordPress core functionality), so it probably makes sense to report there ... however, the classic editor is going away and I'm therefore unsure how much traction it would get (afaik, when applying tags through the new block editor a different endpoint is used and this issue is avoided).

Given that, it may be best to work around this on a site-by-site basis (since I imagine relatively few sites will have this number of tags and this sort of distribution of tags). One approach might be to tweak the minimum number of characters needed to trigger a search, via the `term_search_min_chars` filter (which defaults to 2, but perhaps increasing it to 3 or 4 would somewhat address this in your case)?
